src/main/java/juke/parsers/DateTimeParser.java

```diff
@@ -0,0 +1,86 @@
+package juke.parsers;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Pattern;
+
+import juke.commons.classes.JukeObject;
+import juke.exceptions.parsers.JukeDateFormatParseException;
+
+/**
+ * Parses Datetime Strings into the relevant {@code LocalDateTime} objects.
+ */
+public class DateTimeParser extends JukeObject {
+    /** Regex for recognising DateTime inputs. */
+    private static final String DATETIME_REGEX = "^(0?[1-9]|[12][0-9]|3[01])(\\/|-)(0?[1-9]|1[0-2])(\\/|-)\\d{4} "
+            + "([01]?[0-9]|2[0-3])?(-|:)?[0-5][0-9]$";
+
+    /** Regex for recognising Date inputs. */
+    private static final String DATE_REGEX = "^(0?[1-9]|[12][0-9]|3[01])(\\/|-)(0?[1-9]|1[0-2])(\\/|-)\\d{4}";
+
+    /**
+     * Checks if the input date is of the correct Date format.
+     *
+     * @param date Input date
+     * @return true if the String is a valid Date format, else false
+     */
+    public static boolean isValidDate(String date) {
+        return Pattern.matches(DateTimeParser.DATE_REGEX, date);
+    }
+
+    /**
+     * Checks if the input date is of the correct DateTime format.
+     *
+     * @param datetime Input datetime
+     * @return true if the String is a valid DateTime format, else false
+     */
+    public static boolean isValidDateTime(String datetime) {
+        return Pattern.matches(DateTimeParser.DATETIME_REGEX, datetime);
+    }
+
+    /**
+     * Parses the input String into a {@code LocalDateTime} object, provided that the
+     * string meets the requirement of either being a valid Date or DateTime.
+     *
+     * @param datetime Input date/datetime string
+     * @return {@code LocalDateTime} object representing the input datetime
+     */
+    public static LocalDateTime parse(String datetime) {
+        // strip the datetime first to avoid any errors
+        datetime = datetime.strip();
+
+        if (isValidDateTime(datetime)) {
+            String[] dateTimeSplit = datetime.split(" ");
+            String[] date = dateTimeSplit[0].split("\\/|-");
+            String[] time = Pattern.matches("\\d{4}", dateTimeSplit[1])
+                    ? new String[]{dateTimeSplit[1].substring(0, 2), dateTimeSplit[1].substring(2, 4)}
+                    : dateTimeSplit[1].split("-|:");
```

Comment:
Perhaps this portion can be neater by defining the condition, consequent & alternative first so that it is less complicated
e.g.
```
boolean boolName = Pattern.matches("\\d{4}", dateTimeSplit[1]);
and so on
then
String[] time = boolName ? --- : ---
```

Reply:
Ah I see, I will look into making this chunk of code more readable in a future commit!